### PR TITLE
fix: re-render the list on mark all visited click

### LINKED ISSUE #57: Fix reactivity of "Mark all as visited" event
Now when the user clicks on "Mark all as visited" button the URLs are marked as visited but the UI is not updated.

### PR DESCRIPTION
**Pull request by zvonimirr**
Closes #57

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -91,7 +94,7 @@ export default function Home() {
     }
 
     setHighestPriority(highestPriority);
-    setFeedArchive(storage as FeedArchiveType);
+    setFeedArchive(() => storage as FeedArchiveType);
```
Comment (strdr4605) [inline pages/index.tsx:97]: 🤔 I did not find any explanation for using callback instead of actual value. Maybe you can explain?

**Review by strdr4605 — APPROVED** [review]

File: pages/index.tsx (hunk 1 above)
Comment (zvonimirr) [reply pages/index.tsx:97]: Sure thing:
https://stackoverflow.com/questions/42038590/when-to-use-react-setstate-callback#:~:text=so%20if%20you%20want%20to%20perform%20an%20action%20immediately%20after%20setting%20state%20on%20a%20state%20variable%20and%20then%20return%20a%20result%2C%20a%20callback%20will%20be%20useful

Comment (strdr4605) [reply pages/index.tsx:97]: Ok, but I don't understand the difference in our case `setFeedArchive(storage)` vs `setFeedArchive(() => storage)`

Comment (zvonimirr) [reply pages/index.tsx:97]: I did it with the callback to make sure it happens immediately

Comment (strdr4605) [reply pages/index.tsx:97]: I had a discussion with my teammates about this and we still don't understand the difference.

> to make sure it happens immediately

both `setFeedArchive(storage)`and `setFeedArchive(() => storage)` update the state at the same time (as I understand😅).

--- 

I did a test with `setFeedArchive(storage)` and I see no difference in behavior in UX.

---

I am happy to debate this and see if there is a difference between
`setState(newState)` and `setState(() => newState)`.

Maybe we can ask on some forums?🤔

Comment (zvonimirr) [reply pages/index.tsx:97]: `updateFeeds` is re-declared on every re-render. I'll wrap it in a useCallback instead then

**Pull request merged**